tests/test_utils.py

```diff
@@ -1748,6 +1748,10 @@ def test_iter_by_file(obj, expected):
             [],
             [],
         ),
+        (
+            [[], []],
+            [],
+        ),
```

Comment:
Could you add tests for an empty dict and a dict containing empty dicts as well.

Reply:
According to the type string we do not support passing an (empty) dict, but I extended the tests with the case of a list that contains empty dict(s).